scalanet/src/io/iohk/scalanet/peergroup/kademlia/KRouter.scala

```diff
@@ -129,49 +137,68 @@ class KRouter[A](
     routerState.get.map(_.nodeRecords)
   }
 
-  def ping(recToPing: NodeRecord[A]): Task[Boolean] = {
+  def ping(recToPing: NodeRecord[A]): Task[Option[NodeRecord[A]]] = {
     network
       .ping(recToPing, Ping(uuidSource(), config.nodeRecord))
-      .map(_ => true)
-      .onErrorHandle(_ => false)
-
+      .map(x => Some(x.nodeRecord))
+      .onErrorHandle(_ => None)
   }
 
+  private def isFreshResult(previewsRegisterNodeRecord: Option[NodeRecord[A]], nodeRecord: NodeRecord[A]): Boolean =
+    previewsRegisterNodeRecord.isDefined && previewsRegisterNodeRecord.get.seq > nodeRecord.seq
```

Comment:
1. You can use option `exists` here.
2. I do not folow protocol here. Could leave somewhere docs, about sequence number - what validation should be done on them, how to respond to them, when sequence number is fresh enough, how to generete own sequnce number etc.